extras/src/main/scala/spire/math/extras/interval/IntervalSeq.scala

```diff
@@ -0,0 +1,687 @@
+package spire.math.extras.interval
+
+import java.util.Arrays
+
+import spire.algebra.{Bool, Eq, Order}
+import spire.math.interval._
+import spire.math.{Interval, Rational}
+
+import scala.annotation.tailrec
+import scala.language.implicitConversions
+import scala.reflect.ClassTag
+
+final class IntervalSeq[T] private (
+    val belowAll: Boolean,
+    private val values: Array[T],
+    private val kinds: Array[Byte],
+    private implicit val order: Order[T]) extends IntervalSet[T, IntervalSeq[T]] { lhs =>
+
+  import IntervalSeq._
+
+  private def binarySearch(key: T): Int = {
```

Comment:
(Alternately, if this method performs better than `spire.math.Searching.seach` we should patch that one.)